cmd/minishift/cmd/root.go

```diff
@@ -83,25 +74,22 @@ var RootCmd = &cobra.Command{
 			isAddonInstallRequired bool
 		)
 
-		if !filehelper.Exists(constants.MakeMiniPath("addons")) {
+		constants.MachineName = constants.ProfileName
+		constants.Minipath = constants.GetProfileHomeDir()
+		constants.KubeConfigPath = filepath.Join(constants.Minipath, "machines", constants.MachineName+"_kubeconfig")
+
+		// Initilize the instance directory structure
+		minishiftConfig.InstanceDirs = minishiftConfig.NewMinishiftDirs(constants.ProfileName)
+
+		if !filehelper.Exists(minishiftConfig.InstanceDirs.Addons) {
 			isAddonInstallRequired = true
 		}
 
-		for _, path := range dirs {
-			if err := os.MkdirAll(path, 0777); err != nil {
-				glog.Exitf("Error creating minishift directory: %s", err)
-			}
-		}
+		// creating all directories for minishift run
+		createDirForMinishiftRun(minishiftConfig.InstanceDirs)
```

Comment:
What is a 'minishift run' ?

Reply:
@gbraad We need these directories for running any minishift command.

Reply:
Confused about the name. What is a 'Minishift run'?

Reply:
`m version` doesn't need, so I believe this is instance or profile related.

Reply:
Changing the name to createMinshiftDirs